### PR TITLE
fix(derive_code_mappings): Not all frames have a filename

### PR DESCRIPTION
**Pull request by armenzg**
Fixes [SENTRY-WRF](https://sentry.io/organizations/sentry/issues/3781488456)

### CODE REVIEW

File: src/sentry/tasks/derive_code_mappings.py [hunk 1]
```diff
@@ -90,15 +90,15 @@ def identify_stacktrace_paths(data: NodeData) -> List[str]:
     """
     Get the stacktrace_paths from the event data.
     """
-    if data["platform"] not in SUPPORTED_LANGUAGES:
-        return []
```
Comment (armenzg) [inline src/sentry/tasks/derive_code_mappings.py:94]: This already in place before calling the function:
https://github.com/getsentry/sentry/blob/4ea1fdfcefc4b18f39345f5e6fe2612e7d37c516/src/sentry/tasks/derive_code_mappings.py#L55-L59

File: tests/sentry/tasks/test_derive_code_mappings.py [hunk 2]
```diff
@@ -21,35 +22,44 @@ def setUp(self):
         )
         self.project = self.create_project(organization=self.organization)
 
+    def generate_data(self, frames: List[Dict[str, Union[str, bool]]], platform: str = ""):
```
Comment (armenzg) [inline tests/sentry/tasks/test_derive_code_mappings.py:25]: Helper function. Eventually the Python class should also use it.

File: tests/sentry/tasks/test_derive_code_mappings.py [hunk 3]
```diff
@@ -21,35 +22,44 @@ def setUp(self):
...
-                        "module": "app/utils/handleXhrErrorResponse",
-                        "filename": "./app/utils/handleXhrErrorResponse.tsx",
-                        "abs_path": "webpack:///./app/utils/handleXhrErrorResponse.tsx",
-                        "in_app": True,
-                    },
-                ]
-            },
-        }
-        self.event_data = self.store_event(data=self.test_data, project_id=self.project.id).data
+        self.platform = "javascript"
```
Comment (armenzg) [inline tests/sentry/tasks/test_derive_code_mappings.py:33]: This is the value used inside of `generate_data` as a fallback.

File: tests/sentry/tasks/test_derive_code_mappings.py [hunk 4]
```diff
@@ -21,35 +22,44 @@ def setUp(self):
...
         assert stacktrace_paths == ["./app/utils/handleXhrErrorResponse.tsx"]
 
+    def test_find_stacktrace_paths_bad_data(self):
+        data = self.generate_data([{}])
+        data["stacktrace"]["frames"] = [
+            {
+                "abs_path": "https://example.com/static/chunks/foo.bar.js",
+                "data": {"sourcemap": "https://example.com/_next/static/chunks/foo.bar.js.map"},
+                "in_app": True,
+            }
```
Comment (armenzg) [inline tests/sentry/tasks/test_derive_code_mappings.py:58]: This represents the values from the Sentry error:
image

File: tests/sentry/tasks/test_derive_code_mappings.py [hunk 5]
```diff
@@ -21,35 +22,44 @@ def setUp(self):
...
+        ]
+        stacktrace_paths = identify_stacktrace_paths(data)
```
Comment (armenzg) [inline tests/sentry/tasks/test_derive_code_mappings.py:60]: We now won't raise `KeyError`

**Review by snigdhas — APPROVED** [review]

**Pull request merged**